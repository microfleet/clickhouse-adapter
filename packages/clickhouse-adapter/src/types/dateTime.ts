export class DateTime {
  private readonly value: string

  constructor(timeZone: string) {
    this.value = `DateTime('${timeZone}')`
  }

  public toString() {
    return this.value
  }
}
