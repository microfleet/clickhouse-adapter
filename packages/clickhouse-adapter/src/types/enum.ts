interface EnumProps {
  name: string
  id: string
}

export class EnumClickhouse {
  private readonly value: string

  constructor(enums: EnumProps[]) {
    this.value = enums
      .map((item: { name: string; id: string }) => `'${item.name}' = ${item.id}`)
      .join(',')
  }

  public toString(): string {
    return this.value
  }
}
