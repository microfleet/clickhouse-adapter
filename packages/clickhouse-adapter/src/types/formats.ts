export type Formats = 'JSON' | 'JSONEachRow' | 'CSV' | 'TabSeparated'
