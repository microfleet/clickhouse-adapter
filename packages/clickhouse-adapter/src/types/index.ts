export * from './dateTime'
export * from './enum'
export * from './columnTypes'
