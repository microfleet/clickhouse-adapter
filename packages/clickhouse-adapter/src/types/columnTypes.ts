import { DateTime } from './dateTime'

export type ColumnTypes =
  | 'Int8'
  | 'Int16'
  | 'Int32'
  | 'Int64'
  | 'UInt8'
  | 'UInt16'
  | 'UInt32'
  | 'UInt64'
  | 'Float32'
  | 'Float64'
  | 'String'
  | 'UUID'
  | 'Date'
  | 'DateTime'
  | DateTime
  | 'IPv4'
  | 'IPv6'
