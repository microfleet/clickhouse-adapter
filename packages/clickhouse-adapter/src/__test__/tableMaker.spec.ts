import { DateTime } from '../types'
import { TableMaker } from '..'

describe('Table maker', () => {
  it('create table by fluent API', () => {
    const maker = new TableMaker('ch_test', 'table_test')
    const sql = maker
      .columnDefinition('trackDate', 'Date')
      .columnDefinition('trackTimestamp', 'DateTime')
      .columnDefinition('event_type', 'String')
      .tableOption('ENGINE = MergeTree(trackDate, (event_type), 8192)')
      .toSql()

    expect(sql).toMatchSnapshot()
  })

  it('create table by spec', () => {
    const maker = new TableMaker('ch_test', 'table_test', null, {
      columnDefinitions: [
        { name: 'trackDate', type: 'Date' },
        { name: 'trackTimestamp', type: new DateTime('Europe/Moscow') },
        { name: 'event_type', type: 'String' },
      ],
      tableOptions: ['ENGINE = MergeTree(trackDate, (event_type), 8192)'],
    })

    expect(maker.toSql()).toMatchSnapshot()
  })
})
