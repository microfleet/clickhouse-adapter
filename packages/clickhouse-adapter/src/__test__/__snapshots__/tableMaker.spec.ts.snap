// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table maker create table by fluent API 1`] = `
"CREATE TABLE IF NOT EXISTS ch_test.table_test (
trackDate Date,
trackTimestamp DateTime,
event_type String
) ENGINE = MergeTree(trackDate, (event_type), 8192)

"
`;

exports[`Table maker create table by spec 1`] = `
"CREATE TABLE IF NOT EXISTS ch_test.table_test (
trackDate Date,
trackTimestamp DateTime('Europe/Moscow'),
event_type String
) ENGINE = MergeTree(trackDate, (event_type), 8192)

"
`;
